atomix/cluster/src/main/java/io/atomix/raft/storage/system/MetaStore.java

```diff
@@ -76,6 +80,17 @@ public MetaStore(final RaftStorage storage) throws IOException {
       configurationChannel =
           FileChannel.open(confFile.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
     }
+
+    // Read existing meta info and rewrite with the current version
+    initializeMetaBuffer();
```

Comment:
What is this for?

Reply:
Term and vote are loaded and written independently. Hence, if the version on the disk is different from the current version, it might corrupt the data if you store a new term with new version, but never store the vote or the other way around. So here it rewrites the existing meta info with the new version. It does not matter now, but would be required if we change the version.